cobalt/browser/cobalt_content_browser_client.cc

```diff
@@ -295,4 +327,126 @@ bool CobaltContentBrowserClient::WillCreateURLLoaderFactory(
   return true;
 }
 
+void CobaltContentBrowserClient::CreateExperimentConfig() {
+  if (!exp_config_) {
+    auto pref_registry = base::MakeRefCounted<PrefRegistrySimple>();
+
+    RegisterPrefs(pref_registry.get());
+
+    base::FilePath path;
+    CHECK(base::PathService::Get(content::SHELL_DIR_USER_DATA, &path));
+    path = path.Append(kExperimentConfigFilename);
+
+    PrefServiceFactory pref_service_factory;
+    pref_service_factory.set_user_prefs(
+        base::MakeRefCounted<JsonPrefStore>(path));
+
+    exp_config_ = pref_service_factory.Create(pref_registry);
+  }
+}
+
+void CobaltContentBrowserClient::CreateFeatureListAndFieldTrials() {
+  CreateExperimentConfig();
+  SetUpFieldTrials();
+}
+
+void CobaltContentBrowserClient::SetUpCobaltFeaturesAndParams(
+    std::unique_ptr<base::FeatureList>& feature_list) {
+  // All Cobalt features are associated with the same field trial. This is for
+  // easier feature param lookup.
+  base::FieldTrial* cobalt_field_trial = base::FieldTrialList::CreateFieldTrial(
+      kCobaltExperimentName, kCobaltGroupName);
+  CHECK(cobalt_field_trial) << "Unexpected name conflict.";
+
+  const base::Value::Dict& feature_map =
+      exp_config_->GetDict(kExperimentConfigFeature);
+  const base::Value::Dict& param_map =
+      exp_config_->GetDict(kExperimentConfigFeatureParams);
+
+  for (const auto kvpair : feature_map) {
```

Comment:
Note that here and in l.382, the compiler will copy each std::pair<> iterated over. If that's not needed, then `const auto& kvpair` would be more efficient. It's uncommon to suffix the type ("pair") to variables (this is a Windows practice). So perhaps `const auto& feature_name_and_value` would be better.

Reply:
Naming sgtm. But By adding a reference, compiler complains loop variable binds to a temporary value produced by a range of type 'const base::Value::Dict'. Probably because the reference of const base::Value::Dict goes out of scope by the end of line instead of after the entire for loop?